safe-global/safe-wallet-web
# [Counterfactual] Show dialog with on-ramp options when pressing buy crypto

## What it solves

Part of #3207

## How this PR fixes it

- This PR updates the `Buy Crypto` button to open a Dialog with more (external) options
- Adds a new event `SELECT_ONRAMP_APP` where the label is the name of the app

Note: The `AddFundsCTA` component still needs to be added to the dashboard for new/counterfactual safes

## How to test it

1. Open any Safe
2. Observe a Buy Crypto button is always visible on the dashboard or on an empty assets page
3. Press the button
4. Observe a new Dialog opens with more on-ramp options

## Screenshots
<img width="640" alt="Screenshot 2024-02-09 at 15 04 35" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/473369b1-b93f-4cb3-aedf-5fb5365283ec">
<img width="639" alt="Screenshot 2024-02-09 at 15 04 56" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/bb39c74a-3ef0-420a-aafd-619872923356">
<img width="642" alt="Screenshot 2024-02-09 at 15 05 05" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/9b3ee8e1-72e1-45cc-b4e3-81ab060afe8b">
<img width="422" alt="Screenshot 2024-02-09 at 15 05 16" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/f2b84b9c-61ca-410e-aaea-9ea18d861282">

## Checklist
